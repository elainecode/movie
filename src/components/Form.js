import React from 'react'

const Form = () => {
    return (
      <>
        <div>
          This is the Search Form title container
        </div>
        <div>
        This is the Search Form
      </div>
      <div>
      This is the Search Form button container
    </div>
    </>
    )
}


export default Form