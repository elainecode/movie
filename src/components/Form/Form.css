.form-container {
    color: #fff;
    font-size: small;
    font-weight: 500;
    width: 100%;
    display: grid;
    grid-template-columns: repeat(8, 1fr);
    grid-template-rows: auto auto auto;
    grid-gap: 1em;
    grid-template-areas: 
        "form-title form-title . . . . . ."
        "form-input form-input  form-input  form-input  form-input  form-input form-input ."
        "form-button-a form-button-b form-button-c . . form-button-d form-button-d form-buttom-d";
}
.form-title {
    grid-area: form-title;
}

.form-input {
    grid-area: form-input;
}

.form-button-a { grid-area: form-button-a;}
.form-button-b { grid-area: form-button-b;}
.form-button-c { grid-area: form-button-c;}
.form-button-d { grid-area: form-button-d;}

.align {
    margin: auto;
}

.title-text {
    font-weight: 700;
}
.form-style {
    width: 80%;
}
.form-style input,
.form-style span {
    color: #fff;
}