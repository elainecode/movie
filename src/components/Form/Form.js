import React from 'react';
import { withRouter } from 'react-router-dom';
import {
  FormControl,
  InputLabel,
  Input,
  InputBase,
  FilledInput,
  Button,
  TextField,
  Typography,
  InputAdornment,
  Icon,
} from '@material-ui/core';
import './Form.css';

const Form = ({ updateSearchQuery, searchFilms }) => {
  return (
    <>
      <form onSubmit={searchFilms} className="form-style">
        <InputBase
          fullWidth
          placeholder="Search for a movie"
          onChange={updateSearchQuery}
          inputProps={{ 'aria-label': 'search' }}
          startAdornment={
             <InputAdornment position="start">  <Icon>search</Icon></InputAdornment>
          }
        />
      </form>
    </>
  );
};

export default withRouter(Form);
