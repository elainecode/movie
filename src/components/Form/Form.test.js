import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Form from './Form';

describe('<Form/>', () => {
  test('renders the component', () => {
    const updateSearchQuery = jest.fn();
    const searchFilms = jest.fn();
    const component = mount(
      <Router
        render={routerProps => (
          <Form
            updateSearchQuery={updateSearchQuery}
            searchFilms={searchFilms}
            film={film}
            {...routerProps}
          />
        )}
      />,
    );

    expect(toJson(component)).toMatchSnapshot();

    component.unmount();
  });
});
