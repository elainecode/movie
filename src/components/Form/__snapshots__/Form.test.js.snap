// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form/> renders the component 1`] = `
<BrowserRouter
  render={[Function]}
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  />
</BrowserRouter>
`;
